src/bvals/comms/bnd_info.cpp

```diff
@@ -128,15 +133,28 @@ Indexer6D CalcSetIndices(const NeighborIndexes &ni, LogicalLocation loc,
       }
       ++off_idx;
     } else if (block_offset[dir] > 0) {
-      s[dir] = bounds[dir].e + 1;
+      s[dir] = bounds[dir].e + 1 - top_offset[dir];
       e[dir] = bounds[dir].e + ghosts;
     } else {
       s[dir] = bounds[dir].s - ghosts;
-      e[dir] = bounds[dir].s - 1;
+      e[dir] = bounds[dir].s - 1 + top_offset[dir];
```

Comment:
Same as above: has the offset been added because this element now "owns"?

Reply:
Yep, same explanation as above.